langchain/src/embeddings/voyage.ts

```diff
@@ -0,0 +1,157 @@
+import { chunkArray } from "../util/chunk.js";
+import { getEnvironmentVariable } from "../util/env.js";
+import { Embeddings, EmbeddingsParams } from "./base.js";
+
+enum VoyageModelNames {
+  voyage01 = "voyage-01",
+  voyagelite01 = "voyage-lite-01"
+}
+
+/**
+ * Interface that extends EmbeddingsParams and defines additional
+ * parameters specific to the VoyageEmbeddings class.
+ */
+export interface VoyageEmbeddingsParams extends EmbeddingsParams {
+  modelName: VoyageModelNames;
+
+  /**
+   * The maximum number of documents to embed in a single request. This is
+   * limited by the Voyage AI API to a maximum of 8.
+   */
+  batchSize?: number;
+}
+
+/**
+ * Interface for the request body to generate embeddings.
+ */
+export interface CreateVoyageEmbeddingRequest {
+  /**
+   * @type {string}
+   * @memberof CreateVoyageEmbeddingRequest
+   */
+  model: VoyageModelNames;
+
+  /**
+   *  Text to generate vector expectation
+   * @type {CreateEmbeddingRequestInput}
+   * @memberof CreateVoyageEmbeddingRequest
+   */
+  input: string | string[];
+}
+
+/**
+ * A class for generating embeddings using the Voyage AI API.
+ */
+export class VoyageEmbeddings
+  extends Embeddings
+  implements VoyageEmbeddingsParams
+{
+  modelName = VoyageModelNames.voyage01;
+
+  batchSize = 8;
```

Comment:
This seems a bit small?

Reply:
https://docs.voyageai.com/embeddings/#via-http-request 

> as you can see: input - A single string or a list of strings. The maximum number of strings is 8.